source/validate_cfg.cpp

```diff
@@ -408,45 +392,83 @@ spv_result_t PerformCfgChecks(ValidationState_t& _) {
              << _.getIdName(function.get_id());
     }
 
-    // Updates each blocks immediate dominators
+    // Prepare for dominance calculations.  We want to analyze all the
+    // blocks in the function, even in degenerate control flow cases
+    // including unreachable blocks.  For this calculation, we create an
+    // agumented CFG by adding a pseudo-entry node that is considered the
+    // predecessor of each source in the original CFG, and a pseudo-exit
+    // node that is considered the successor to each sink in the original
+    // CFG.  The augmented CFG is guaranteed to have a single source node
+    // (i.e. not having predecessors) and a single exit node (i.e. not
+    // having successors).  However, there might be isolated strongly
+    // connected components that are not reachable by following successors
+    // from the pseudo entry node, and not reachable by following
+    // predecessors from the pseudo exit node.
+
+    // We need vectors to use as the predecessors (in the augmented CFG)
+    // for the source nodes of the original CFG.  It must have a stable
+    // address for the duraction of the calculation.
+    auto* pseudo_entry = function.get_pseudo_entry_block();
+    auto* pseudo_exit = function.get_pseudo_exit_block();
+    const vector<BasicBlock*> pseudo_entry_vec = {pseudo_entry};
+    // Similarly, we need a vector to be used as the successors (in the
+    // augmented CFG) for sinks in the original CFG.
+    const vector<BasicBlock*> pseudo_exit_vec = {pseudo_exit};
+    // Returns the predecessors of a block in the augmented CFG.
+    auto augmented_predecessor_fn =
+        [pseudo_entry, &pseudo_entry_vec](const BasicBlock* block) {
+          auto predecessors = block->get_predecessors();
+          return (block != pseudo_entry && predecessors->empty())
+                     ? &pseudo_entry_vec
```

Comment:
That's 100% deliberate.  I can also work around it by having that particular vector be served up by the Function object.